[MCP] Introduce an experimental official rust sdk based mcp client

gpeal opened the pull request:
The [official Rust SDK](https://github.com/modelcontextprotocol/rust-sdk/tree/57fc428c578a1a3fe851ee0838bf068bda120eb3) has come a long way since we first started our mcp client implementation 5 months ago and, today, it is much more complete than our own stdio-only implementation.

This PR introduces a new config flag `experimental_use_rmcp_client` which will use a new mcp client powered by the sdk instead of our own.

To keep this PR simple, I've only implemented the same stdio MCP functionality that we had but will expand on it with future PRs.

gpeal committed:
Works

gpeal committed:
logging client handler + stderr test

gpeal committed:
use_*

gpeal committed:
Doc

AdsQnn commented:
Either way, the code should roughly look like this:

```
let npx_path = which("npx")?; // works like the `which` command
let status = Command::new(npx_path)
    .arg("-v")
    .status()
    .await?;
```


Without which, it won’t work on Windows.

gpeal committed:
Added an e2e test

gpeal committed:
WIP integration test

pakrym-oai committed:
test: limit mock sse response to once and add debug logs

Limit mock SSE response to a single use in tests; add debug eprintlns to rmcp_client tests. Fix order of edition in Cargo.toml and enable tokio io-std feature. Move stdio helper into rmcp_test_server, and comment out ServerInfo name/version fields.

gpeal committed:
Cleanup

gpeal committed:
Merge remote-tracking branch 'origin/main' into gpeal/rmcp-client-1

gpeal committed:
Fix sandbox check

gpeal committed:
Add some logs

gpeal committed:
Logs for failing test

gpeal commented on codex-rs/core/tests/suite/rmcp_client.rs:
```diff
@@ -0,0 +1,162 @@
+
+    let expected_env_value = "propagated-env";
+    let rmcp_test_server_bin = CargoBuild::new()
+        .package("codex-rmcp-client")
```
I had to use escargot instead of the existing cargo crate because this binary is in the rmcp-client crate

gpeal committed:
PR review

gpeal committed:
Merge remote-tracking branch 'origin/main' into gpeal/rmcp-client-1
